### PR TITLE
Add BART-L text summarization notebook

### PR DESCRIPTION
**Pull request by jayniep-gc**
# What does this PR do?

* Adds the BART-L text summarisation notebook.
* Updates README to include new notebook.
* Fixes minor typos in README.

### CODE REVIEW

File: notebooks/bart/pytorch/tests/test_notebook_text_summarization.py
Comment (katalinic-gc) [inline notebooks/bart/pytorch/tests/test_notebook_text_summarization.py:1]: I think including this test will be more work than necessary if the goal is just to land the notebook.

For more context, we don't have notebook tests at present in `optimum.graphcore`, so for this test to be usable we'd need to add a test harness akin to `tests/test_examples.py` which creates the virtual env around it etc (I am assuming that the `ExecutePreprocessor` will run the `pip install` steps as well, but I'm not familiar with it beyond documentation). Doing all this is probably more suited for a separate PR.

File: notebooks/bart/pytorch/notebook_text_summarization.ipynb [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "executable_cache_dir = os.getenv(\"POPLAR_EXECUTABLE_CACHE_DIR\", \"/tmp/exe_cache/\")"
```
Comment (katalinic-gc) [inline notebooks/bart/pytorch/notebook_text_summarization.ipynb:78]: You are not using this, to enable executable caching it has to be provided to the `ipu_config` below as a `executable_cache_dir` kwarg.

File: notebooks/bart/pytorch/notebook_text_summarization.ipynb
Comment (katalinic-gc) [inline notebooks/bart/pytorch/notebook_text_summarization.ipynb:1]: The notebook should be moved to the `notebooks` parent folder, to mimic what is done for the majority of other notebooks.

File: notebooks/bart/pytorch/README.md
[the remark this answers is not in the document]
Comment (akrzgc) [reply notebooks/bart/pytorch/README.md:1]: removed README

File: notebooks/bart/pytorch/tests/test_notebook_text_summarization.py
Comment (akrzgc) [reply notebooks/bart/pytorch/tests/test_notebook_text_summarization.py:1]: removed the test

File: notebooks/bart/pytorch/notebook_text_summarization.ipynb [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+    "\n",
+    "summarizer = pipeline(\n",
+    "    \"summarization\",\n",
+    "    model=\"facebook/bart-large-cnn\",\n",
+    "    tokenizer=\"facebook/bart-large-cnn\",\n",
+    "    ipu_config=ipu_config.to_dict(),\n",
+    "    config=\"facebook/bart-large-cnn\",\n",
+    "    max_input_length=1024,\n",
+    "    truncation=True,\n",
+    "    max_length=150,\n",
```
[the remark this answers is not in the document]
Comment (akrzgc) [reply notebooks/bart/pytorch/notebook_text_summarization.ipynb]: removed the `max_length=150` arg

File: notebooks/bart/pytorch/notebook_text_summarization.ipynb
Comment (akrzgc) [reply notebooks/bart/pytorch/notebook_text_summarization.ipynb:1]: moved the notebook to parent folder and renamed to `text_summarization.ipynb`

File: notebooks/bart/pytorch/notebook_text_summarization.ipynb (hunk 1 above)
Comment (akrzgc) [reply notebooks/bart/pytorch/notebook_text_summarization.ipynb]: good catch, thanks, fixed

File: notebooks/text_summarization.ipynb [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+    "Copyright (c) 2023 Graphcore Ltd. All rights reserved."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Text Summarization on IPUs using BART-L - Inference\n",
+    "\n",
+    "This notebook demonstrates a text summarization task with a BART-L model using an inference pipeline from 🤗 Optimum, run on Graphcore IPUs.\n",
```
Comment (katalinic-gc) [inline notebooks/text_summarization.ipynb:16]: Optimum Graphcore

File: notebooks/text_summarization.ipynb [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+    "summarizer.model.detachFromDevice()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Conclusions and next steps\n",
+    "\n",
+    "This notebook demonstrated running a text summarization task on Graphcore IPUs, with BART-L using an inference pipeline from 🤗 Optimum.\n",
```
Comment (katalinic-gc) [inline notebooks/text_summarization.ipynb:318]: Ditto

**Review by katalinic-gc — APPROVED** [review]

File: notebooks/text_summarization.ipynb (hunk 4 above)
Comment (akrzgc) [reply notebooks/text_summarization.ipynb]: done

File: notebooks/text_summarization.ipynb (hunk 3 above)
Comment (akrzgc) [reply notebooks/text_summarization.ipynb]: done

File: notebooks/text_summarization.ipynb [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from optimum.graphcore import pipeline\n",
+    "\n",
+    "summarizer = pipeline(\n",
+    "    \"summarization\",\n",
+    "    model=\"facebook/bart-large-cnn\",\n",
+    "    tokenizer=\"facebook/bart-large-cnn\",\n",
+    "    ipu_config=ipu_config.to_dict(),\n",
```
Comment (katalinic-gc) [inline notebooks/text_summarization.ipynb:124]: Last nit, you can pass the config directly, doesn't need to be converted to a dict.

**Pull request merged**